tests/foreman/virtwho/ui/test_esx.py

```diff
@@ -51,6 +53,29 @@ def form_data():
 
 
 class TestVirtwhoConfigforEsx:
+    def create_http_proxy(self, name=None, url=None, type='https'):
```

Comment:
@hkx303 we can use api to create http proxy that'd faster.
```
        http_proxy = entities.HTTPProxy(
            name=gen_string('alpha', 15),
            url=settings.http_proxy.auth_proxy_url,
            username=settings.http_proxy.username,
            password=settings.http_proxy.password,
            organization=[org.id],
        ).create()
```
- There are many tests which needs http_proxy so probably we should create an api_fixture and start using it in tests. You can add fixture in `pytest_fixtures/api_fixtures.py`
- We already have CLI and UI tests for Http Proxy so those should be enough to test CRUD operations.

Note: There is change in password field, so I have raised https://github.com/SatelliteQE/airgun/pull/542 to fix it.

Reply:
- Use API instead of UI. 
- Have not added it in `pytest_fixtures/api_fixtures.py` because I saw several uses of it in other cases but maybe only our cases need to create HTTP and HTTPS proxy and return the URL for now.